tokio-console/src/view/task.rs

```diff
@@ -68,76 +67,172 @@ impl TaskView {
             })
             .collect();
 
+        let stats_area_check = Layout::default()
+            .direction(layout::Direction::Horizontal)
+            .constraints(
+                [
+                    layout::Constraint::Percentage(50),
+                    layout::Constraint::Percentage(50),
+                ]
+                .as_ref(),
+            )
+            .split(area);
+
+        let location_heading = "Location: ";
+        let location_max_width = stats_area_check[0].width as usize - 2 - location_heading.len(); // NOTE: -2 for the border
+        let max_width_stats_area = area.width - 45;
+        let mut location_lines_vector: Vec<String> = task
+            .location()
+            .to_string()
+            .chars()
+            .collect::<Vec<char>>()
+            .chunks(max_width_stats_area as usize)
+            .map(|chunk| chunk.iter().collect())
+            .collect();
+        let no_of_lines_extra_required_to_accomadate_location = location_lines_vector.len() - 1;
         let (
             controls_area,
             stats_area,
             poll_dur_area,
             scheduled_dur_area,
             fields_area,
             warnings_area,
-        ) = if warnings.is_empty() {
-            let chunks = Layout::default()
-                .direction(layout::Direction::Vertical)
+        ) = if task.location().len() > location_max_width {
```

Comment:
We definitely don't want to add another layer of `if-else` statements here. I think that be specifying the task stats height outside, we can leave this like it was (just depending on the presense of warnings).